src/tests/BuildWasmApps/Wasm.Build.Tests/ICUShardingTests.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Abstractions;
+using System.Globalization;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class ICUShardingTests : BuildTestBase
+    {
+        public ICUShardingTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+            : base(output, buildContext)
+        {
+        }
+
+        public static IEnumerable<object?[]> ICUShardingTestData(bool enableSharding, RunHost host)
+            => ConfigWithAOTData(false)
+                .Multiply(
+                    new object?[] { enableSharding, "en", "en-GB", false},      //EFIGS
+                    new object?[] { enableSharding, "en", "zh", true},          //EFIGS
+                    new object?[] { enableSharding, "es-ES", "am-ET", true},    //EFIGS
+                    new object?[] { enableSharding, "es", "fr-FR", true},       //EFIGS
+                    new object?[] { enableSharding, "zh", "zh-Hans", false},    //CJK
+                    new object?[] { enableSharding, "ko", "zh-Hans", false},    //CJK
+                    new object?[] { enableSharding, "ja-JP", "es-ES", true},    //CJK //this one should have thrown
+                    new object?[] { enableSharding, "am-ET", "de-DE", false},   //no_CJK
+                    new object?[] { enableSharding, "am-ET", "ko", true},       //no_CJK
+                    new object?[] { enableSharding, "am-ET", "ja-JP", true},    //no_CJK
+                    new object?[] { enableSharding, "am-ET", "vi-VN", false},   //no_CJK
+                    new object?[] { enableSharding, null, "zh-Hans", true})     //ALL
+                .WithRunHosts(host)
+                .UnwrapItemsAsArrays();
+
+        [Theory]
+        [MemberData(nameof(ICUShardingTestData), parameters: new object[] { true, RunHost.All })]
```

Comment:
Should there be one for `enableSharding==false`?

Reply:
With sharding turned off, all of the cultures should be loaded successfully. `System.Globalization.Tests` should handle all of those cases. So really I guess don't even need an `enableSharding` switch.

Reply:
No harm in testing it here too.